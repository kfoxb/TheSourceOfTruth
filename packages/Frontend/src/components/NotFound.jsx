import React from 'react';
import styled from 'styled-components';
import Divider from '@material-ui/core/Divider';
import Button from '@material-ui/core/Button';
import colors from '../constants/colors';

const StyledDiv = styled.div`
  background-color: ${colors.darkGrey};
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  height: 100vh;
`;

const StyledHeader = styled.div`
  text-align: center;
  font-size: 50px;
  color: ${colors.white}
  font-family: 'Noto Sans', sans-serif;
  position: absolute;
  margin: auto;
  right: 0;
  left: 0;
  top: 25%;
`;

export default function NotFound() {
  return (
    <StyledDiv>
      <StyledHeader>
        <p style={{ fontSize: '150px', margin: '0 0 0em' }}>404</p>
        <p style={{ fontSize: '25px' }}>Oops, something went wrong</p>
        <Divider style={{ backgroundColor: `${colors.white}` }} />
        <p style={{ margin: '0 0 0em' }}>PAGE NOT FOUND</p>
        <Button variant="raised" style={{ backgroundColor: `${colors.blue}`, color: `${colors.white}`, top: '10px' }}>
          Return to Home
        </Button>
      </StyledHeader>
    </StyledDiv>
  );
}
