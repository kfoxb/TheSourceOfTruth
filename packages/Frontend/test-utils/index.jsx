/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
