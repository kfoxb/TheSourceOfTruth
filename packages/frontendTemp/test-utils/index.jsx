/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
