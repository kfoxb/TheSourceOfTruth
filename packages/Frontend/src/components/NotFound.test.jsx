/* eslint-env jest */
import { createAssertWithPropsToMatchSnapshot } from '../../test-utils';
import NotFound from './NotFound';

describe('NotFound', () => {
  it('should render', () => {
    createAssertWithPropsToMatchSnapshot(NotFound.WrappedComponent, {})();
  });
});
