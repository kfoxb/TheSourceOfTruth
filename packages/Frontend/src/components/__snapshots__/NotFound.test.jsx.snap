// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render 1`] = `
<Route
  render={[Function]}
/>
`;
